waspc/cli/Command.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module Command
+    ( Command
+    , runCommand
+    , CommandError(..)
+    ) where
+
+import Control.Monad.Except (MonadError, ExceptT, runExceptT)
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Catch (MonadThrow)
+
+
+newtype Command a = Command { _runCommand :: ExceptT CommandError IO a }
+    deriving (Functor, Applicative, Monad, MonadIO, MonadError CommandError, MonadThrow)
+
+runCommand :: Command a -> IO ()
+runCommand cmd = do
+    errorOrResult <- runExceptT $ _runCommand cmd
+    case errorOrResult of
+        Left cmdError -> putStrLn $ "Error: " ++ _errorMsg cmdError
+        Right _ -> return ()
+
+-- TODO: What if we want to recognize errors in order to handle them?
+--   Should we add _commandErrorType? Should CommandError be parametrized by it, is that even possible?
```

Comment:
Yes, I guess we could. Although, we might want that all of them have this "msg" so that we can always easily display them hm! So in that case it might make sense to start with this type we have now and then nest a sum type under it.
Hm hm, let's see how it evolves. From my experience, usually to handle an error it is enough to have message + type of error, for 90% cases, but ok that depends on the use case of course.